src/main/java/com/nike/cerberus/client/HttpClientFactory.java

```diff
@@ -0,0 +1,192 @@
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import com.nike.cerberus.ConfigConstants;
+import com.nike.cerberus.domain.environment.CertificateInformation;
+import com.nike.cerberus.module.CerberusModule;
+import com.nike.cerberus.store.ConfigStore;
+import com.nike.vault.client.StaticVaultUrlResolver;
+import com.nike.vault.client.auth.DefaultVaultCredentialsProviderChain;
+import okhttp3.OkHttpClient;
+import okio.Buffer;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Named;
+import javax.net.ssl.*;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class HttpClientFactory {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    public static final int DEFAULT_TIMEOUT = 15;
+    public static final TimeUnit DEFAULT_TIMEOUT_UNIT = TimeUnit.SECONDS;
+
+    private final ObjectMapper objectMapper;
+    private final ConfigStore configStore;
+
+    @Inject
+    public HttpClientFactory(@Named(CerberusModule.CONFIG_OBJECT_MAPPER) ObjectMapper objectMapper,
+                             ConfigStore configStore) {
+
+        this.objectMapper = objectMapper;
+        this.configStore = configStore;
+    }
+
+    /**
+     * Admin client for doing admin cms tasks
+     */
+    public CerberusAdminClient createCerberusAdminClient(String url) {
+        return new CerberusAdminClient(
+                new StaticVaultUrlResolver(url),
+                new DefaultVaultCredentialsProviderChain(),
+                new OkHttpClient.Builder()
+                        .hostnameVerifier(new NoopHostnameVerifier())
+                        .connectTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                        .writeTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                        .readTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                        .build(),
+                objectMapper
+        );
+    }
+
+    /**
+     * @return Generic default client with timeouts for making manual http calls
+     */
+    public OkHttpClient getGenericClient() {
+        return new OkHttpClient.Builder()
+                .hostnameVerifier(new NoopHostnameVerifier())
+                .connectTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .writeTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .readTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .build();
+    }
+
+    /**
+     * Downloads the CA Chains from S3 for the Cerberus certs and creates a client that can talk to the
```

Comment:
This only appears necessary if you want to talk to individual CMS instances, but it is nice to not have to worry about installing the CA, think CI systems with tons of nodes.